sway-fmt-v2/src/items/item_fn.rs

```diff
@@ -1,15 +1,206 @@
 use crate::{
-    fmt::{Format, FormattedCode, Formatter},
-    FormatterError,
+    config::items::ItemBraceStyle,
+    fmt::{Format, FormattedCode, Formatter, FormatterError},
+    utils::bracket::{CurlyBrace, Parenthesis},
 };
-use sway_parse::ItemFn;
+use std::fmt::Write;
+use sway_parse::{token::Delimiter, CodeBlockContents, FnArg, FnArgs, FnSignature, ItemFn};
+use sway_types::Spanned;
 
 impl Format for ItemFn {
     fn format(
         &self,
-        _formatted_code: &mut FormattedCode,
+        formatted_code: &mut FormattedCode,
+        formatter: &mut Formatter,
+    ) -> Result<(), FormatterError> {
+        self.fn_signature.format(formatted_code, formatter)?;
+        Self::open_curly_brace(formatted_code, formatter)?;
+        self.body
+            .clone()
+            .into_inner()
+            .format(formatted_code, formatter)?;
+        Self::close_curly_brace(formatted_code, formatter)?;
+
+        Ok(())
+    }
+}
+
+// For now this just pushes the char
```

Comment:
```suggestion
// For now this just pushes the char
```
Is this comment still relevant, I feel like you have implemented logic for placing the where as well? Maybe we can get rid of this if it is not.

Reply:
I'll get that out in my next PR (: